src/Components/Web.JS/src/Services/NavigationUtils.ts

```diff
@@ -43,6 +44,22 @@ export function isWithinBaseUriSpace(href: string) {
   && (nextChar === '' || nextChar === '/' || nextChar === '?' || nextChar === '#');
 }
 
+export function hasProgrammaticEnhancedNavigationHandler(): boolean {
+  return programmaticEnhancedNavigationHandler !== undefined;
+}
+
+export function attachProgrammaticEnhancedNavigationHandler(handler: typeof programmaticEnhancedNavigationHandler) {
+  programmaticEnhancedNavigationHandler = handler;
+}
+
+export function performProgrammaticEnhancedNavigation(absoluteInternalHref: string, replace: boolean): void {
+  if (!programmaticEnhancedNavigationHandler) {
+    throw new Error('No enhanced programmatic navigation handler has been attached');
+  }
+
+  programmaticEnhancedNavigationHandler(absoluteInternalHref, replace);
+}
```

Comment:
This dance is to avoid `NavigationEnhancement.ts` and `NavigationManager.ts` from importing each other. That way, `blazor.server.js` and `blazor.webassembly.js` don't pay the cost of SSR features that they don't use.